config/rbac/namespaced/role.yaml

```diff
@@ -0,0 +1,168 @@
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  creationTimestamp: null
+  name: manager-role
+rules:
+- apiGroups:
+  - sagemaker.aws.amazon.com
+  resources:
+  - batchtransformjobs
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - patch
+  - delete
+- apiGroups:
```

Comment:
What is our plan to keep these updated? Normally, kubebuilder generators produce new CRD roles [from annotation](https://github.com/aws/amazon-sagemaker-operator-for-k8s/blob/master/controllers/trainingjob/trainingjob_controller.go#L72-L73) so we don't have to worry about it.

We should have either automated way to do it or documentation, this will have to be done for any new k8s resource. If you don't already have a better way to do it, I think the generators can be configured to write to this Kustomize overlay dependency, or namespaced scope kustomize overlays can depend on normal CRD overlay (which is already setup with generators).

Reply:
Yeah I might have to write patches for things that are auto-generated by kubebuilder. I will run the generator and amend anything it produces - problem may be with the default paths it is installed into. Kustomize requires resources exist at the same level or deeper (which really messes with how I can organise the folders).

Reply:
Ok, wow that is an annoying constraint by Kustomize. Ideally we don't have to do anything manual to do CRD updates, hopefully that's easy to implement

Reply:
ok, this is a bit a repelling,  if we have to manually update the CRD then it will cause issue in future. For developer that experience might be not good. what all options we have here ?

Reply:
Turns out it actually wasn't too hard to change. Just had to manage my directories a little bit, but I removed code duplication fairly easily and just created a patch to change ClusterRole to Role when generating the namespace scoped installers.